### PR TITLE
Publish Schema for ApiManagement 2019-12-01

### PR DESCRIPTION
**Pull request by solankisamir**
- add schema at relevant places
- add tests

### CODE REVIEW

**Comment by solankisamir** [comment]
@anthony-c-martin can you take a look?

**Comment by anthony-c-martin** [comment]
@solankisamir this schema has already been generated by https://github.com/Azure/azure-resource-manager-schemas/blob/master/schemas/2019-12-01/Microsoft.ApiManagement.json - ApiManagement has been [onboarded for automatic generation](https://github.com/Azure/azure-resource-manager-schemas/tree/master/generator#onboarding-teams-to-autogeneration-pipeline).

Note that going forwards we no longer require individual schema tests for teams that are onboarded.

File: schemas/2019-04-01/deploymentTemplate.json [hunk 1]
```diff
@@ -1277,7 +1277,47 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2019-07-01/Microsoft.Compute.json#/resourceDefinitions/virtualMachineScaleSets_virtualmachines" },
                   { "$ref": "https://schema.management.azure.com/schemas/2019-07-01/Microsoft.Compute.json#/resourceDefinitions/virtualMachineScaleSets_virtualMachines_extensions" },
                   { "$ref": "https://schema.management.azure.com/schemas/2019-07-01/Microsoft.Compute.Extensions.json#/resourceDefinitions/virtualMachines_extensions" },
-                  { "$ref": "https://schema.management.azure.com/schemas/2019-07-01/Microsoft.Compute.Extensions.json#/resourceDefinitions/virtualMachineScaleSets_extensions" }
+                  { "$ref": "https://schema.management.azure.com/schemas/2019-07-01/Microsoft.Compute.Extensions.json#/resourceDefinitions/virtualMachineScaleSets_extensions" },
```
Comment (anthony-c-martin) [inline schemas/2019-04-01/deploymentTemplate.json:1280]: This schema is already being referenced by this file - please undo this change.

**Comment by solankisamir** [comment]
Closing this PR, as APIM is onboarded automatic generation of schema
